cms/djangoapps/contentstore/rest_api/v0/views/tabs.py

```diff
@@ -0,0 +1,239 @@
+""" API Views for course tabs """
+
+import edx_api_doc_tools as apidocs
+from django.utils.translation import ugettext_lazy as _
+from opaque_keys.edx.keys import CourseKey
+from rest_framework import status
+from rest_framework.request import Request
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from xmodule.modulestore.django import modulestore
+from xmodule.modulestore.exceptions import ItemNotFoundError
+
+from common.djangoapps.student.auth import has_studio_read_access, has_studio_write_access
+from openedx.core.lib.api.view_utils import DeveloperErrorViewMixin, verify_course_exists, view_auth_classes
+from ..serializers import CourseTabSerializer, CourseTabUpdateSerializer, TabIDLocatorSerializer
+from ....views.tabs import edit_tab_handler, get_course_tabs, reorder_tabs_handler
+
+
+@view_auth_classes(is_authenticated=True)
+class CourseTabListView(DeveloperErrorViewMixin, APIView):
+    """
+    API view to list course tabs.
+    """
+
+    @apidocs.schema(
+        parameters=[apidocs.string_parameter("course_id", apidocs.ParameterLocation.PATH, description="Course ID")],
+        responses={
+            200: CourseTabSerializer,
+            401: "The requester is not authenticated.",
+            403: "The requester cannot access the specified course.",
+            404: "The requested course does not exist.",
+        },
+    )
+    @verify_course_exists
+    def get(self, request: Request, course_id: str) -> Response:
+        """
+        Get a list of all the tabs in a course including hidden tabs.
+
+        **Example Request**
+
+            GET /api/contentstore/v0/tabs/{course_id}
+
+        **Response Values**
+
+        If the request is successful, an HTTP 200 "OK" response is returned.
+
+        The HTTP 200 response contains a list of objects that contain info
+        about each tab.
+
+        **Example Response**
+
+        ```json
+        [
+            {
+                "course_staff_only": false,
+                "is_hidden": false,
+                "is_hideable": false,
+                "is_movable": false,
+                "name": "Home",
+                "settings": {},
+                "tab_id": "info",
+                "title": "Home",
+                "type": "course_info"
+            },
+            {
+                "course_staff_only": false,
+                "is_hidden": false,
+                "is_hideable": false,
+                "is_movable": false,
+                "name": "Course",
+                "settings": {},
+                "tab_id": "courseware",
+                "title": "Course",
+                "type": "courseware"
+            },
+            ...
+        }
+        ```
+        """
+        course_key = CourseKey.from_string(course_id)
+        if not has_studio_read_access(request.user, course_key):
```

Comment:
In the course advance settings, we allowed authenticated users to access the `get` calls but we are restricting in the case of course tabs. is it intentional?

Reply:
That is not the case, there too we check for studio read access.

Reply:
missed it, thank.s